src/app/cluster/details/external-cluster/external-cluster-delete-confirmation/template.html

```diff
@@ -0,0 +1,56 @@
+<!--
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<km-dialog-title>Delete Cluster</km-dialog-title>
+<mat-dialog-content>
+  <p>Delete
+    <b (click)="copy(cluster.name)"
+       matTooltip="Click to copy"
+       class="km-copy">{{cluster.name}}</b>
+    cluster permanently?
+  </p>
+
+  <mat-form-field>
+    <mat-label>Cluster Name</mat-label>
+    <input id="km-delete-cluster-dialog-input"
+           required
+           matInput
+           type="text"
+           title="Cluster name"
+           (blur)="onChange($event)"
+           (keyup)="onChange($event)"
+           #clusterNameInput>
+  </mat-form-field>
+
+
+  <div *ngIf="hasMachineDeployments"
+       fxLayoutAlign=" start"
+       class="warning-container">
+    <i class="km-icon-warning"></i>
+    <div class="km-text dynamic-kubelet-config-warning">
+      {{errorMessage}}
+    </div>
+  </div>
+</mat-dialog-content>
+
+<mat-dialog-actions>
+  <km-button id="km-delete-cluster-dialog-delete-btn"
+             icon="km-icon-delete"
+             label="Delete Cluster"
+             [disabled]="inputName !== cluster.name || !hasMachineDeployments"
```

Comment:
I don't have enough knowledge here but are we intentionally blocking external cluster deletion until there is at least 1 machine deployment? I think in normal (non-external) clusters we can delete the cluster immediately after creation.

Reply:
correct:

In **External Clusters** this is cloud specific as far i know cluster can ONLY be deleted when NO **nodegroups** are attached.

We are just showing warning to user if `at least 1 machine deployment?` than user can't delete that cluster from cloud provider.

**Edge Cases:**

**Case-1:**
- New cluster created and user tries to delete  -> Warning appears as MD is set to this cluster but Cloud Provider takes time to initialise it so it is not in **Active** state -> User can be able to delete that cluster.

**Case-2:**
- MD exists something wrong with in a cluster and Cluster overall is in Healthy state but MD is not user will see warning but cluster can be deleted.